clients/admin-ui/src/features/custom-fields/CustomFieldsTable.tsx

```diff
@@ -0,0 +1,93 @@
+import { Box, Flex, Spinner, Text } from "@fidesui/react";
+import { useHasPermission } from "common/Restrict";
+import { FidesTable, TitleCell, WrappedCell } from "common/table";
+import EmptyTableState from "common/table/EmptyTableState";
+import React, { useMemo } from "react";
+import { Column } from "react-table";
+
+import { useAppSelector } from "~/app/hooks";
+import {
+  selectAllCustomFieldDefinitions,
+  useGetAllCustomFieldDefinitionsQuery,
+} from "~/features/plus/plus.slice";
+import { CustomFieldDefinitionWithId, ScopeRegistryEnum } from "~/types/api";
+
+import { EnableCustomFieldCell, FieldTypeCell } from "./cells";
+
+export const CustomFieldsTable = () => {
+  const { isLoading } = useGetAllCustomFieldDefinitionsQuery();
+  const customFields = useAppSelector(selectAllCustomFieldDefinitions);
+
+  // Permissions
+  const userCanUpdate = useHasPermission([
+    ScopeRegistryEnum.CUSTOM_FIELD_UPDATE,
+  ]);
+  const columns: Column<CustomFieldDefinitionWithId>[] = useMemo(
+    () => [
+      {
+        Header: "Title",
+        accessor: "name",
+        Cell: TitleCell,
+      },
+      {
+        Header: "Description",
+        accessor: "description",
+        Cell: WrappedCell,
+      },
+      { Header: "Field Type", accessor: "field_type", Cell: FieldTypeCell },
+      { Header: "Locations", accessor: "resource_type", Cell: WrappedCell },
+      {
+        Header: "Enable",
+        accessor: (row) => !row.active,
+        disabled: !userCanUpdate,
+        Cell: EnableCustomFieldCell,
+      },
+    ],
+    [userCanUpdate]
+  );
+
+  if (isLoading) {
+    return (
+      <Flex height="100%" justifyContent="center" alignItems="center">
+        <Spinner />
+      </Flex>
+    );
+  }
+
+  if (customFields.length === 0) {
+    return (
+      <EmptyTableState
+        title="It looks like it’s your first time here!"
+        buttonHref=""
+        buttonText="Add a custom field"
+        description={
+          <>
+            You haven’t created any custom fields yet. To create a custom field,
+            click on the <strong>&quot;Add a custom field&quot;</strong> button
+          </>
+        }
+      />
+    );
+  }
+
+  return (
+    <>
```

Comment:
nit: doesn't look like you need the `<>` wrapper here

Reply:
I tried removing it but it complains saying that JSX elements need a least one parent element

Reply:
Is there another way to fix that issue?

Reply:
oh sorry, I had thought it was all contained in the one `<Box>` and didn't realize there were actually two sibling `<Box>`s. in which case yes, we do need the `<>` :)